hypershift-operator/metrics.go

```diff
@@ -485,6 +530,71 @@ func (m *hypershiftMetrics) observeNodePools(ctx context.Context, nodePools *hyp
 	return nil
 }
 
+func getOperatorImage(client crclient.Client) error {
+	ctx := context.TODO()
+	hypershiftNamespace := os.Getenv("MY_NAMESPACE")
+	hypershiftPodName := os.Getenv("MY_NAME")
+
+	hypershiftPod := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: hypershiftPodName, Namespace: hypershiftNamespace}}
+	if err := client.Get(ctx, crclient.ObjectKeyFromObject(hypershiftPod), hypershiftPod); err != nil {
+		if errors.IsNotFound(err) {
+			hypershiftDeployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: hypershiftNamespace, Name: assets.HypershiftOperatorName}}
+			if err := client.Get(ctx, crclient.ObjectKeyFromObject(hypershiftDeployment), hypershiftDeployment); err != nil {
+				return err
+			}
+			for _, c := range hypershiftDeployment.Spec.Template.Spec.Containers {
+				if c.Name == assets.HypershiftOperatorName {
+					image = c.Image
+					imageId = "not found"
+				}
+			}
+		} else {
+			return err
+		}
+	} else {
+		for _, c := range hypershiftPod.Status.ContainerStatuses {
+			if c.Name == assets.HypershiftOperatorName {
+				image = c.Image
+				imageId = c.ImageID
+			}
+		}
+	}
+	return nil
+}
+
+// getLastTransitionToAvailable returns the seconds when successfully got a Running operator pod
+// until we have the fully Ready status in the pod's status.conditions
+func getLastTransitionToAvailable(client crclient.Client) func() float64 {
+	return func() float64 {
+		ctx := context.TODO()
+		hypershiftNamespace := os.Getenv("MY_NAMESPACE")
+		hypershiftPodName := os.Getenv("MY_NAME")
+		var runningAt time.Time
+		var lastTransitionToReady time.Time
+
+		hypershiftPod := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: hypershiftPodName, Namespace: hypershiftNamespace}}
+		if err := client.Get(ctx, crclient.ObjectKeyFromObject(hypershiftPod), hypershiftPod); err != nil {
+			return float64(0)
+		}
+		// Get when the pod get in Running state
+		for _, c := range hypershiftPod.Status.ContainerStatuses {
+			if c.Name == assets.HypershiftOperatorName {
+				runningAt = c.State.Running.StartedAt.Time
+			}
+		}
+
+		// Get the last transition time to Ready state
+		for _, c := range hypershiftPod.Status.Conditions {
+			if c.Type == corev1.PodReady {
+				lastTransitionToReady = c.LastTransitionTime.Time
+			}
+		}
+
+		// return the difference
+		return float64(lastTransitionToReady.Sub(runningAt).Seconds())
```

Comment:
why do we need this duration?

Reply:
This is the number of seconds from the Starting Pod until the Ready status. I think that was mentioned by @csrwng in a previous review cycle.

Reply:
I think returning c.LastTransitionTime.Time is probably what we want here